### PR TITLE
consolidate kibana api to internal project structure

### PR DESCRIPTION
**Pull request by adam-stokes**
Add all relevant kibana api calls into the golang project standard of
`internal/kibana`.

Signed-off-by: Adam Stokes <51892+adam-stokes@users.noreply.github.com>

## What does this PR do?

This PR is the start of some refactoring to get the project in a more standardized layout. It adds all the related kibana api calls into `internal/kibana` and also has additional api calls added to abstract out more of the work that is being handled in the testsuites. Currently, this is only an additive PR to get the bits in place for a further refactor of the testsuites and cli.

## Why is it important?

It is nice to have all the relevant api calls under a single location that is easily identifiable

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
~~- [ ] I have run the Unit tests for the CLI, and they are passing locally~~
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
~~- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)~~

## How to test this PR locally

Nothing to test at this moment as there are several subsequent PR's that will utilize this new api in the scenario testsuites

## Related issues

- Relates #959 

## Follow-ups

Once this PR is approved and merge I have several additional changes that make use of this code in the testsuites.

This is part of a larger refactor https://github.com/elastic/e2e-testing/compare/feat-kibana-refactor that gives a better idea of where these new additions will be utilized and which of the original code is moved/removed. Most of the code related to this PR lives in `cli/services/kibana.go` and additional code lives in `e2e/_suites/fleet/fleet.go` `e2e/_suites/fleet/integrations.go`. In the referenced `feat-kibana-refactor` branch you can get a better idea of how that looks and get a feel for the direction I am taking.

### CODE REVIEW

**Comment by cachedout** [comment]
I'm a bit confused by the word "moving" in the description of this PR. I see files being added but nothing being moved or otherwise deleted. Is this PR purely additive or is it a refactor? Perhaps a description of where this code or logic is being moved _from_ would make things more clear. Thanks!

**Comment by adam-stokes** [comment]
> I'm a bit confused by the word "moving" in the description of this PR. I see files being added but nothing being moved or otherwise deleted. Is this PR purely additive or is it a refactor? Perhaps a description of where this code or logic is being moved _from_ would make things more clear. Thanks!

@cachedout Thanks! I updated the PR with some additional context, let me know if you have any questions

**Comment by mdelapenya** [comment]
IIRC there are unit tests for the client. If possible please keep them (with updates) during the refactor

File: internal/kibana/agents.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Jeffail/gabs/v2"
+	log "github.com/sirupsen/logrus"
+)
+
+// UnEnrollAgent unenrolls agent from fleet
+func (c *Client) UnEnrollAgent(hostname string, force bool) error {
```
Comment (mdelapenya) [inline internal/kibana/agents.go:16]: NIT: could you reorder the methods alphabetically when possible? That would help when searching through the file

Comment (mdelapenya) [reply internal/kibana/agents.go:16]: I think that once this is done, we can merge this PR 😉

**Comment by adam-stokes** [comment]
Will close this in favor of #1008

**Comment by mdelapenya** [comment]
Closing as per https://github.com/elastic/e2e-testing/pull/992#issuecomment-816380834

**Pull request closed without merging**